src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -911,16 +917,59 @@ private function createRequestMatcher(ContainerBuilder $container, string $path
             return $this->requestMatchers[$id];
         }
 
-        // only add arguments that are necessary
-        $arguments = [$path, $host, $methods, $ips, $attributes, null, $port];
-        while (\count($arguments) > 0 && !end($arguments)) {
-            array_pop($arguments);
+        $arguments = [];
+        if ($methods) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([MethodRequestMatcher::class, $methods]), MethodRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$methods])
+            ;
+            $arguments[] = new Reference($lid);
+        }
+
+        if ($path) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([PathRequestMatcher::class, $path]), PathRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$path])
+            ;
+            $arguments[] = new Reference($lid);
+        }
+
+        if ($host) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([HostRequestMatcher::class, $host]), HostRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$host])
+            ;
+            $arguments[] = new Reference($lid);
+        }
+
+        if ($ips) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([IpsRequestMatcher::class, $ips]), IpsRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$ips])
+            ;
+            $arguments[] = new Reference($lid);
+        }
+
+        if ($attributes) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([AttributesRequestMatcher::class, $attributes]), AttributesRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$attributes])
+            ;
+            $arguments[] = new Reference($lid);
+        }
+
+        if ($port) {
+            $container->register($lid = $id.'.'.ContainerBuilder::hash([PortRequestMatcher::class, $port]), PortRequestMatcher::class)
+                ->setPublic(false)
+                ->setArguments([$port])
+            ;
+            $arguments[] = new Reference($lid);
         }
 
         $container
-            ->register($id, RequestMatcher::class)
+            ->register($id, ChainRequestMatcher::class)
```

Comment:
```suggestion
       $container->register($id, ChainRequestMatcher::class)
```
?

Reply:
I'd keep the current name: it doesn't have the same interface + needless BC break ahead when renaming.